lib/oauth/factory.go

```diff
@@ -359,14 +363,18 @@ func WithCookiePrefix(prefix string) Modifier {
 
 func WithFlags(fl *Flags) Modifier {
 	return func(o *Options) error {
-		if len(fl.OauthSecretJSON) == 0 && (fl.OauthSecretID == "" || fl.OauthSecretKey == "") {
-			return fmt.Errorf("you must specify the secret-file or (secret-key and secret-id) options")
+		var mods []Modifier
+		if len(fl.AdditionalOAuth) == 0 {
```

Comment:
is this logic still needed?

Reply:
yes, it will still error out due to ordering if this isnt there

Reply:
AdditionalOauth -> should be renamed JsonConfig.